Added plotting feature to data downloader

AntWarland opened the pull request:
See demo of plotting function and downloader below.

Let me know if you run into issues running it - I definitely had to install one or two conda packages. I've attached my conda packages list if that helps but it should prompt for anything it needs. Please could you make a note of any you install as they will be needed for the live website (sorry to have not done this myself).

[packages in environment.txt](https://github.com/user-attachments/files/20839501/packages.in.environment.txt)

[Screencast from 2025-06-20 16-18-32.webm](https://github.com/user-attachments/assets/40912af8-d3b6-4344-984c-7b15d01beaf7)

AntWarland commented:
@SamAnderson27 Just added my environment.yml to the env directory.

SamAnderson27 commented:
Hi Ant, having looked at this it seems to be working fine and looks really good.
The yaml file is really messy and could do with some cleaning up if you have some time when you're back!
If it's possible could you change the headers in the csv files to something that's a bit more descriptive? If it's just us using this page then it will be fine for us to interpret but if the clinical/tech teams try to track some data it won't make any sense to them - although they'll probably just ask us to get the data anyway - might be worth looking into.
Other than this great work!

AntWarland commented:
- I've added plotly to the main.yaml (turns out that's all it needed) and deleted environment.yaml. Tested a fresh install of the env and it works fine.
- Re: changing then csv headers, I agree they're clunky but as it will be primarily us that's using this I'm inclined to keep them as they are for now as the data model prefix effectively assigns a category to each field, making them easier to find in the plot variable dropdowns and navigate the csv columns (in my opinion). Can change if needed though.
- Made a couple of other quality of life code changes. Should be good to merge now.

SamAnderson27 commented:
Hi Ant, looks really good! I'll merge this into Dev now!
